src/QirRuntime/public/ChunkedStore.hpp

```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include <assert.h>
+#include <vector>
+
+#include "BitStates.hpp"
+
+namespace Microsoft
+{
+namespace Quantum
+{
+    /*==============================================================================

```

Comment:
Why is this part of the public API? It seems like an internal implementation class.

Reply:
Initially I set up the project with two folders for headers: public and internal (in addition to the headers that are consumed only inside their own folder) but got feedback from Dimitry and Sasha that setup like this tends to cause maintainence issues and that there is no harm in over-sharing utilities headers.